logging/google/cloud/logging/handlers/transports/background_thread.py

```diff
@@ -48,14 +50,21 @@ def _get_many(queue_, max_items=None):
     :param max_items: The maximum number of items to get. If ``None``, then all
         available items in the queue are returned.
 
+    :type min_wait_time: float
+    :param min_wait_time: The minimum number of seconds to wait for items from
+        a queue.
+
     :rtype: Sequence
     :returns: A sequence of items retrieved from the queue.
     """
+    start = time.time()
     # Always return at least one item.
     items = [queue_.get()]
     while max_items is None or len(items) < max_items:
         try:
-            items.append(queue_.get_nowait())
+            elapsed = time.time() - start
+            timeout = max(0, min_wait_time - elapsed)
+            items.append(queue_.get(timeout=timeout))
```

Comment:
This actually sets the "maximum wait time", if there are items in the queue this will still return immediately until there are none but will also wait an additional amount of time specified by `timeout` to see if any more items come in. If that's the intended behavior, cool. I would name this `max_latency` to match PubSub's [batch semantics](https://github.com/GoogleCloudPlatform/google-cloud-python/blob/master/pubsub/google/cloud/pubsub_v1/publisher/batch/thread.py#L86).

Reply:
If `max_items is None`, this function will always take at least `min_wait_time` seconds to run. Example:

* `min_wait_time = 2`
* queue item 1 is there at the start
* queue item 2 comes after 1s
* queue item 3 comes after 3s

The function starts, and immediately `items == [item1]`. It then enters the loop and calls `queue_.get(timeout=2)`. After 1s, `items == [item1, item2]`. It does another iteration and calls `queue_.get(timeout=1)`. This will time out, and the function returns `[item1, item2]`. Only one second after the function returns, queue item 3 will be enqueued.

Reply:
@jonparrott You are right, however, in that if `max_items` is set, this function will return earlier if `min_wait_time` is set and `max_items` has been reached. I am happy to rename this to `max_latency` if you feel that this is more intuitive.

Reply:
Yeah, I think `max_latency` is better and fits with the Pub/Sub strategy. Basically pubsub does the "same" thing as this - collects some messages in the background and sends them to the server after a certain amount of time has passed.

Reply:
I've renamed the variable to `max_latency`, as suggested.